core-plugins/docs/Joiner-batchjoiner.md

```diff
@@ -74,6 +74,23 @@ Setting it to true can cause a large drop in performance if there are a lot of n
 **Number of Partitions:** Number of partitions to use when grouping fields. If not specified, the execution
 framework will decide on the number to use.
 
+**Distribution:** Enabling distribution will increase the level of parallelism when joining skewed data. A 
+skewed join is when multiple records from one stage (the skewed stage) join to a single record in
+another stage. Distribution is possible when the following conditions are met:
+1. There are exactly two input stages. 
+1. Broadcast is not enabled for either stage.
+1. The skewed input is marked as `required`.
+
+
+Distribution requires two parameters:
+1. **Distribution Size:** This controls the size of the salt that will be generated for distribution. The **Number of Partitions** 
+property should be greater than or equal to this number for optimal results. A larger value will lead to more 
+parallelism but it will also grow the size of the non-skewed dataset by this factor.
+1. **Skewed Input Stage:**  Name of the skewed input stage. The skewed input stage is the one that contains many rows that join 
+to the same row in the non-skewed stage. Ex. If stage A has 10 rows that join on the same row in stage B, then
+stage A is the skewed input stage.
```

Comment:
Seeing how many issues we see with skew in user pipelines, I think it's worth having a separate section after the example section that talks about what skew is and how to handle it. 

We should start by defining what skew is, use an example to illustrate, and explain why it results in slowness. The first approach to deal with it is to load the smaller side of the join in memory. If that isn't possible, use the distribution properties.

Reply:
done